Orange/base.py

```diff
@@ -22,14 +62,15 @@ class Learner:
     learner_adequacy_err_msg = ''
 
     def __init__(self, preprocessors=None):
-        if preprocessors is None:
-            preprocessors = type(self).preprocessors
-        self.preprocessors = list(preprocessors)
+        self.use_default_preprocessors = False
+        if isinstance(preprocessors, Iterable):
+            self.preprocessors = preprocessors
```

Comment:
Generators are legitimate iterables but will unwind only once. Safest to cast it to tuple explicitly.